packages/css/src/index.js

```diff
@@ -125,28 +125,53 @@ const transforms = [
 export const responsive = styles => theme => {
   const next = {}
   const breakpoints = get(theme, 'breakpoints', defaultBreakpoints)
-  const mediaQueries = [
-    null,
-    ...breakpoints.map(n => `@media screen and (min-width: ${n})`),
-  ]
+  let mediaQueries
+  if(Array.isArray(breakpoints)) {
+    mediaQueries = [null, ...breakpoints.map((n) => `@media screen and (min-width: ${n})`)]
+  }
+  else {
+    mediaQueries = {}
+    for (const key in breakpoints) {
+      mediaQueries[key] = `@media screen and (min-width: ${breakpoints[key]})`
+    }
+  }
 
   for (const key in styles) {
     const value = styles[key]
     if (value == null) continue
-    if (!Array.isArray(value)) {
-      next[key] = value
-      continue
-    }
-    for (let i = 0; i < value.length; i++) {
-      const media = mediaQueries[i]
-      if (value[i] == null) continue
-      if (!media) {
-        next[key] = value[i]
+    if(typeof value === 'object') {
+      if (Array.isArray(value)) {
+        for (const i in value) {
+          const media = mediaQueries[i]
+          if (value[i] == null) continue
+          if (!media) {
+            next[key] = value[i]
+            continue
+          }
+          next[media] = next[media] || {}
+          next[media][key] = value[i]
+        }
+        continue
+      }
+      else {
+        for (const k in value) {
```

Comment:
From an API standpoint at this time, I do *not* want to allow object based syntax for the `style` objects (which are a bit different from props) that the `css` package uses. The array syntax is optional and there is already an object-based approach using native media query syntax, e.g.

```js
css={css({
  width: '100%',
  '@media screen and (min-width: 40em)': {
    width: '50%',
  }
})}
```

I think this adds API surface area in a way that there are two ways to do the same thing, e.g.

```js
{
  width: {
    xsmall: '100%',
    small: '50%',
  },
  '@media screen and (min-width: 40em)': {
    '33%'
  }
}
```

The above approach could be tied into the `theme.breakpoints` scale with some sort of helper for the media query key, which I think would make more sense from an API perspective